openedx_learning/apps/authoring/publishing/api.py

```diff
@@ -1012,6 +1469,11 @@ def _create_container_version(
             title=title,
             created=created,
             created_by=created_by,
+            dependencies=[
+                entity_row.entity_id
+                for entity_row in entity_list.rows
+                if entity_row.is_unpinned()
+            ]
```

Comment:
Checking my understanding.. hypothetically:
* we have a unit U (published and draft @ version Uv1)
* we have component C (never published, draft @ version Cv2)
* Uv1 pins Cv1
* When you render the published version of Uv1, it includes Cv1, even though C is unpublished
* Now, Cv2 is published.
* There is no dependency relationship between Uv1 and C, so this does not affect Uv1
* and thus, we do not consider Uv1 to "contain unpublished changes"
* and, as before, when you render the published version of Uv1, it includes Cv1

right?

Reply:
Yup.